### PR TITLE
Np 717 get all customers

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 import static org.apache.http.HttpStatus.SC_OK;
 
-public class GetCustomerHandler extends ApiGatewayHandler<Customer,Customer> {
+public class GetCustomerHandler extends ApiGatewayHandler<String,Customer> {
```
Comment (janbj) [inline get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java:20]: What do we set when we have no input data?

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package no.unit.nva.customer.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("PMD.ShortMethodName")
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java:6]: Shouldn't we ignore this by default? It does not seem a meaningful rule

File: customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+     * @param customers list of Customer
+     * @return customerList
+     */
+    public static CustomerList of(List<Customer> customers) {
+        CustomerList list = new CustomerList();
+        list.addAll(customers);
+        return list;
+    }
+
+    public static CustomerList of(Customer customers) {
```
Comment (axthosarouris) [inline customer-commons/src/main/java/no/unit/nva/customer/model/CustomerList.java:21]: Make it into a varargs?

File: customer-commons/src/test/java/no/unit/nva/customer/model/CustomerListTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package no.unit.nva.customer.model;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CustomerListTest {
+
+    @Test
+    public void customerListFromCustomer() {
```
Comment (axthosarouris) [inline customer-commons/src/test/java/no/unit/nva/customer/model/CustomerListTest.java:10]: test case where customer is null?

File: get-all-customers/src/test/java/no/unit/nva/customer/getall/GetAllCustomersHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        handler = new GetAllCustomersHandler(customerService, environment);
+        outputStream = new ByteArrayOutputStream();
+        context = new TestContext();
+    }
+
+    @Test
+    public void requestToHandlerReturnsCustomerList() throws Exception {
+        UUID identifier = UUID.randomUUID();
+        Customer customer = new Customer.Builder()
+                .withIdentifier(identifier)
```
Comment (axthosarouris) [inline get-all-customers/src/test/java/no/unit/nva/customer/getall/GetAllCustomersHandlerTest.java:62]: mockCustomerService()

File: get-all-customers/src/test/java/no/unit/nva/customer/getall/GetAllCustomersHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                .build();
+        CustomerList customers = CustomerList.of(customer);
+        when(customerService.getCustomers()).thenReturn(customers);
+
+        Map<String,Object> headers = getRequestHeaders();
+        InputStream inputStream = inputStream(headers);
```
Comment (axthosarouris) [inline get-all-customers/src/test/java/no/unit/nva/customer/getall/GetAllCustomersHandlerTest.java:68]:
```
        Map<String,Object> headers = getRequestHeaders();
        InputStream inputStream = inputStream(headers);
```
could be one method `createRequest()`

File: get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java (hunk 1 above)
Comment (axthosarouris) [reply get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java:20]: In the current version String. But maybe we fix this

File: get-all-customers/src/test/java/no/unit/nva/customer/getall/GetAllCustomersHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    private Map<String, String> getResponseHeaders() {
+        return Map.of(
+                CONTENT_TYPE, APPLICATION_JSON.getMimeType(),
+                ACCESS_CONTROL_ALLOW_ORIGIN, WILDCARD
+        );
+    }
+
+    private Map<String, String> getErrorResponseHeaders() {
```
Comment (axthosarouris) [inline get-all-customers/src/test/java/no/unit/nva/customer/getall/GetAllCustomersHandlerTest.java:98]: This is not in use. Add a test for the error response?

File: create-customer/src/test/java/no/unit/nva/customer/create/CreateCustomerHandlerTest.java [hunk 8]
```diff
@@ -82,13 +82,13 @@ public void requestToHandlerReturnsCustomerCreated() throws Exception {
 
     private Map<String, Object> getRequestHeaders() {
         return Map.of(
-                CONTENT_TYPE, APPLICATION_JSON,
-                ACCEPT, APPLICATION_JSON);
+                CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType(),
```
Comment (brinxmat) [inline create-customer/src/test/java/no/unit/nva/customer/create/CreateCustomerHandlerTest.java:85]: This is repeated in GetAllCustomersHandlerTest.java:85 and GetCustomerHandlerTest.java:119
 maybe split into Util.

File: get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java (hunk 1 above)
Comment (axthosarouris) [inline get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java:20]: We can set Void :) thanks @brinxmat

File: get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java [hunk 9]
```diff
@@ -43,12 +43,12 @@ public GetCustomerHandler() {
      * @param environment   environment
      */
     public GetCustomerHandler(CustomerService customerService, Environment environment) {
-        super(Customer.class, environment);
+        super(String.class, environment);
```
Comment (axthosarouris) [inline get-customer/src/main/java/no/unit/nva/customer/get/GetCustomerHandler.java:46]: Use Void

**Pull request merged**